spec/tapioca/compilers/symbol_table_compiler_spec.rb

```diff
@@ -531,6 +531,37 @@ class WebMockHTTPClient
       assert_equal(output, compile)
     end
 
+    it("compiles a constant which is aliased to a constant that has been overwritten and the names don't match") do
+      add_ruby_file("bar.rb", <<~RUBY)
+        class MyClient
+        end
+
+        module MyModule
+          # The test above doesn't catch this because the names are different
+          OriginalClient = ::MyClient
+        end
+
+        class MockClient < MyClient
+        end
+
+        MyClient = MockClient
+      RUBY
+
+      output = template(<<~RBI)
+        class MockClient
+        end
+
+        MyClient = MockClient
+
+        module MyModule
+        end
+
+        MyModule::OriginalClient = MyClient
```

Comment:
I don't think this is what we want to generate, though. Let us call the first `MyClass` class `MyClass (1)` and the second one `MyClass (2)` to disambiguate them. At runtime after the `bar.rb` file is loaded `MyModule::OriginalClient` would be aliasing `MyClass (1)` and `MyClass (2)` would really be an alias to `MockClient`.

But the RBI we generate is saying that `MyModule::OriginalClient` is an alias to `MyClient (2)` which is an alias to `MockClient`. This does not match the runtime behaviour.

I think that's why we had the `are_equal?` check in the first place.

For this case, I'd prefer to not generate `MyModule::OriginalClient` at all, or to generate it as `MyModule::OriginalClient = Class.new` or `MyModule::OriginalClient = Module.new`, whichever is appropriate.

Reply:
Ah interesting OK. I'll go back to the drawing board 👍